base/cvd/cuttlefish/host/commands/host_bugreport/main.cc

```diff
@@ -180,34 +178,38 @@ Result<void> CvdHostBugreportMain(int argc, char** argv) {
     }
 
     if (FLAGS_include_adb_bugreport) {
-      // TODO(b/359657254) Create the `adb bugreport` asynchronously.
-      std::string device_br_dir = TempDir() + "/cvd_dbrXXXXXX";
-      CF_EXPECTF(mkdtemp(device_br_dir.data()) != nullptr,
-                 "mkdtemp failed: '{}'", strerror(errno));
-      auto result = CreateDeviceBugreport(instance, device_br_dir);
-      if (result.ok()) {
-        auto names = DirectoryContents(device_br_dir);
-        if (names.ok()) {
-          for (const auto& name : names.value()) {
-            std::string filename = device_br_dir + "/" + name;
-            LogError(AddFileAt(archive, filename,
-                               android::base::Basename(filename)));
-          }
-        } else {
-          LOG(ERROR) << "Cannot read from device bugreport directory: "
-                     << names.error().FormatForEnv(/* color = */ false);
-        }
-      } else {
-        LOG(ERROR) << "Failed to create device bugreport: "
-                   << result.error().FormatForEnv(/* color = */ false);
-      }
-      static_cast<void>(RecursivelyRemoveDirectory(device_br_dir));
+      LogError(AddAdbBugreport(instance, archive));
     }
   }
 
   AddNetsimdLogs(archive);
 
   LOG(INFO) << "Building cvd bugreport completed";
+}
+
+Result<void> CvdHostBugreportMain(int argc, char** argv) {
+  ::android::base::InitLogging(argv, android::base::StderrLogger);
+  google::ParseCommandLineFlags(&argc, &argv, true);
+
+  std::string log_filename = TempDir() + "/cvd_hbr.log.XXXXXX";
+  {
+    auto fd = SharedFD::Mkstemp(&log_filename);
+    CF_EXPECT(fd->IsOpen(), "Unable to create log file: " << fd->StrError());
+    android::base::SetLogger(TeeLogger({
+        {ConsoleSeverity(), SharedFD::Dup(2), MetadataLevel::ONLY_MESSAGE},
+        {LogFileSeverity(), fd, MetadataLevel::FULL},
+    }));
+  }
+
+  auto config = CuttlefishConfig::Get();
+  CHECK(config) << "Unable to find the config";
+
+  WritableZip archive = CF_EXPECT(ZipOpenReadWrite(FLAGS_output));
+
+  // This function will gather as much as it can and log what it couldn't. It
+  // won't fail because the bugreport log file needs to be added to the archive
+  // and the archive needs to be finalized.
```

Comment:
"it won't fail because" doesn't sound exactly right here: is "failures are ignored because" more accurate?